FertigMelder/module.php

```diff
@@ -37,51 +37,13 @@ public function ApplyChanges() {
 		//Never delete this line!
 		parent::ApplyChanges();
 		
-		$sourceID = $this->ReadPropertyInteger("SourceID");
-		
-		$eid = @IPS_GetObjectIDByIdent("EventUp", $this->InstanceID);
-		if ($eid == 0) {
-			$eid = IPS_CreateEvent(0);
-			IPS_SetParent($eid, $this->InstanceID);
-			IPS_SetIdent($eid, "EventUp");
-			IPS_SetName($eid, "EventUp");
-			IPS_SetHidden($eid, true);
-			IPS_SetEventTriggerSubsequentExecution($eid, false);
-			IPS_SetEventScript($eid, 'FM_CheckEvent($_IPS[\'TARGET\'], "Up");');
-		}
-		if ($sourceID != 0) {
-			IPS_SetEventTrigger($eid, 2, $sourceID); // Grenzwertunterschreitung
-			IPS_SetEventTriggerValue ($eid, $this->ReadPropertyFloat("BorderValue"));
-			IPS_SetEventActive($eid, false);
-		}
-		
-		$eid = @IPS_GetObjectIDByIdent("EventDown", $this->InstanceID);
-		if ($eid == 0) {
-			$eid = IPS_CreateEvent(0);
-			IPS_SetParent($eid, $this->InstanceID);
-			IPS_SetIdent($eid, "EventDown");
-			IPS_SetName($eid, "EventDown");
-			IPS_SetHidden($eid, true);
-			IPS_SetEventTriggerSubsequentExecution($eid, false);
-			IPS_SetEventScript($eid, 'FM_CheckEvent($_IPS[\'TARGET\'], "Down");');
-		}
-		if ($sourceID != 0) {
-			IPS_SetEventTrigger($eid, 3, $sourceID); // Grenzwertunterschreitung
-			IPS_SetEventTriggerValue ($eid, $this->ReadPropertyFloat("BorderValue"));
-			IPS_SetEventActive($eid, false);
-		}
-		
-		if ($sourceID != 0) {
-			$this->SetActive(GetValue($this->GetIDForIdent("Active")));
-		}
-		
+		$this->RegisterMessage($this->ReadPropertyInteger("SourceID"), VM_UPDATE);
+	
```

Comment:
Es wäre schön, wenn du hier das "alte" Event aufräumen könntest.

//Hier Kommentar
$eid = @IPS_GetObjectIDByIdent("EventUp", $this->InstanceID);	
if ($eid > 0) {
... IPS_DeleteEvent($eid);
}

$eid = @IPS_GetObjectIDByIdent("EventDown", $this->InstanceID);
....